Tests/AzSdk.test.reference.props

```diff
@@ -1,9 +1,9 @@
 <Project ToolsVersion="15.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
 	<Import Project="$([MSBuild]::GetPathOfFileAbove('test.props'))" />
   <ItemGroup>
-    <PackageReference Include="Microsoft.Rest.ClientRuntime.Azure.Authentication" Version="2.3.6" />
     <PackageReference Include="Microsoft.Azure.Test.HttpRecorder" Version="1.12.0" />
     <PackageReference Include="Microsoft.Rest.ClientRuntime.Azure.TestFramework" Version="1.7.4" />
```

Comment:
Why change the references here?

Reply:
Microsoft.Rest.ClientRuntime.Azure.Authentication is upgraded to 2.4.0, and project build fine without explicit version on it.

Microsoft.IdentityModel.Clients.ActiveDirectory is in conflict versioning, so it had to be added as explicit version.
ClientRuntime.Azure.TestFramework 1.7.4 requires < 4.0.0   (even for latest version 1.7.6)
Microsoft.Rest.ClientRuntime.Azure.Authentication 2.4.0 requires >= 4.3.0